### PR TITLE
adding new container instance options

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
This PR is adding several options azure_rm_containerinstance which are essential.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
azure_rm_containerinstance

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py [hunk 1]
```diff
@@ -89,6 +94,38 @@
             ports:
                 description:
                     - List of ports exposed within the container group.
+            environment_variables:
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py:97]: is it an overwrite updating? how to remove an environment variable?

File: lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py [hunk 2]
```diff
@@ -243,6 +284,10 @@ def __init__(self):
                 type='list',
                 required=True
             ),
+            restart_policy=dict(
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py:289]: there's no spec for ports?

File: lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py [hunk 3]
```diff
@@ -362,24 +408,35 @@ def create_update_containerinstance(self):
...
             ports = []
+            variables = []
 
             port_list = container_def.get("ports")
             if port_list:
                 for port in port_list:
                     ports.append(self.cgmodels.ContainerPort(port=port))
 
+            variable_list = container_def.get("environment_variables")
+            if variable_list:
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py:423]: what if update an existing environemtn variable or remove one?

File: lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py (hunk 1 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py:97]: yes, it's overwrite, doesn't work like tags

File: lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py (hunk 3 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py:423]: it's full overwrite

Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py:423]: current container instance implementation has no proper idempotency as well.
this is one of the things that need to be handled properly and it's a large change which I mentioned today.

**Review by yungezz — COMMENTED** [review]
LGTM, but need clearly call  out behavior of environment variable lists overwrite  behavior.

**Review by yungezz — APPROVED** [review]

**Comment by xkrt** [comment]
Can you merge this to 2.7 branch, so it will be released in next 2.7.8?

**Comment by zikalino** [comment]
@xkrt not really... can't merge new features into 2.7.x :-(
just bugfixes

**Pull request merged**